source/server/configuration_impl.cc

```diff
@@ -106,12 +106,7 @@ void MainImpl::initializeTracers(const envoy::config::trace::v2::Tracing& config
 
   // Now see if there is a factory that will accept the config.
   auto& factory = Config::Utility::getAndCheckFactory<HttpTracerFactory>(type);
-  if (factory.requiresClusterName() && server.localInfo().clusterName().empty()) {
```

Comment:
reviewer note: This is the only real code change in here. I removed this check to avoid having to end up implementing a fake trace driver just to test this. When I looked into this, I don't think this check has any real purpose. This information is used only to set tags, and we don't enforce other aspects of local info which are also used to set tags. I think it's OK for them to be empty and then have empty tags. (It also seemed pretty weird that the dynamic trace drivers would not care about this for example.)

Reply:
I opened https://github.com/envoyproxy/envoy/issues/2989 to further look into this.

Reply:
This was added in https://github.com/envoyproxy/envoy/pull/2252

@rnburn since I have less of an understanding of why it was added

Reply:
Yeah, the check is actually not for target cluster, it's actually for the local cluster, which ends up being used in tags (zipkin and LS have other checks for the target cluster existing). I should have caught this during your review but I didn't. I think it's fine to remove this check and we an revisit in the linked ticket if we want to have guards on various "local info" being available.